parser/ddsmtparser.py

```diff
@@ -1125,39 +1122,14 @@ def get_subst (self, node):
             node = self.substs[node.id]
         return node
 
-
-class SMTNodeSubstList (SMTSubstList):
-
-    def subst (self, node, substitution):
-        assert (isinstance (node, SMTNode))
-        super().subst(node, substitution)
-
-
-class SMTScopeSubstList (SMTSubstList):
-
-    def subst (self, node, substitution):
-        assert (isinstance (node, SMTScopeNode))
-        assert (not self.is_subst(node))
-        super().subst(node, substitution)
-
-
-class SMTCmdSubstList (SMTSubstList):
-
-    def subst (self, node, substitution):
-        assert (isinstance (node, SMTCmdNode))
-        assert (not self.is_subst(node))
-        super().subst(node, substitution)
-
-
 class SMTFormula:
+    g_node_id = 0 #id for all three kinds of nodes
```

Comment:
Nit: space after `#`, also maybe improve this to "unique id for nodes of kind SMTNode, SMTCmdNode and SMTScopeNode".